lib/rubocop/ast/node/mixin/method_dispatch_node.rb

```diff
@@ -9,6 +9,7 @@ module MethodDispatchNode
       include MethodIdentifierPredicates
 
       ARITHMETIC_OPERATORS = %i[+ - * / % **].freeze
+      SPECIAL_MODIFIERS = %w[private protected].freeze
```

Comment:
There's also `public`.

Reply:
It maintains the behavior extracted from `Layout/IndentationWidth` cop.

```diff
      class IndentationWidth < Cop # rubocop:disable Metrics/ClassLength
         (snip)
-        SPECIAL_MODIFIERS = %w[private protected].freeze
```

https://github.com/rubocop-hq/rubocop/blob/v0.69.0/lib/rubocop/cop/layout/indentation_width.rb#L55

I think that the purpose is to implement the following style. `public` is not included here.

> Indent and no blank line after private/protected.

https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#follow-the-coding-conventions

Reply:
Hmm, I assume that's just some oversight because `public` is not used very often. I don't see any reason why someone would want to treat it differently from `private` and `protected`.

Reply:
Yes. I agree that `public` is not used very often.

On the other hand, I investigated about `Layout/IndentationWidth` history.
https://github.com/rubocop-hq/rubocop/pull/1759

I think that `private` and `protected` are intended to be treated specially as internal modifiers. For example, a Rails controller code created by `rails new` is:
https://github.com/rails/rails/blob/v6.0.0.rc1/railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb.tt#L53-L66

```ruby
  private
    # Use callbacks to share common setup or constraints between actions.
    def set_<%= singular_table_name %>
      @<%= singular_table_name %> = <%= orm_class.find(class_name, "params[:id]") %>
    end

    # Only allow a trusted parameter "white list" through.
    def <%= "#{singular_table_name}_params" %>
      <%- if attributes_names.empty? -%>
      params.fetch(:<%= singular_table_name %>, {})
      <%- else -%>
      params.require(:<%= singular_table_name %>).permit(<%= permitted_params %>)
      <%- end -%>
    end
```

Perhaps one improvement is to change the constant name `SPECIAL_MODIFIERS` to something like `INTERNAL_MODIFIERS`.

The following behavior changes will occur if `public` is added to constant `SPECIAL_MODIFIERS`.
Therefore I'd like to keep the current behavior first.

## An example

```ruby
% cat example.rb
class Test
  public

  def e
  end

  protected

  def f
  end

  private

  def g
  end
end
```

### Do not include `public` in `SPECIAL_MODIFIERS`

```diff
% rubocop -a
% git diff
diff --git a/example.rb b/example.rb
index 44fe815..d24b0f1 100644
--- a/example.rb
+++ b/example.rb
@@ -1,16 +1,15 @@
+# frozen_string_literal: true
+
 class Test
   public

-  def e
-  end
+  def e; end

   protected

-  def f
-  end
+    def f; end

   private

-  def g
-  end
+    def g; end
 end
```

### Include `public` in `SPECIAL_MODIFIERS`

```diff
% rubocop -a
% git diff
diff --git a/example.rb b/example.rb
index 44fe815..8760788 100644
--- a/example.rb
+++ b/example.rb
@@ -1,16 +1,15 @@
+# frozen_string_literal: true
+
 class Test
   public

-  def e
-  end
+    def e; end

   protected

-  def f
-  end
+    def f; end

   private

-  def g
-  end
+    def g; end
 end
```

Reply:
Hmm, so basically the Rails team doesn't want to outdent stuff under public? 

I guess it might be best to just expose the modifiers as a config, so people would decide if they want to outdent stuff consistently or not. Hardcoding special cases would be a worse solution IMO.